panther_analysis_tool/main.py

```diff
@@ -398,6 +399,90 @@ def upload_analysis(args: argparse.Namespace) -> Tuple[int, str]:
     return 0, ""
 
 
+def confirm_analysis_exists(args: argparse.Namespace, analysis_id_list: list) -> list:
+    validation_payload = {"listDetections": {"ids": analysis_id_list, "fields": ["id"]}}
+    client = get_client(args.aws_profile, "lambda")
+
+    validation = client.invoke(
+        FunctionName="panther-analysis-api",
+        InvocationType="RequestResponse",
+        LogType="None",
+        Payload=json.dumps(validation_payload),
+    )
+
+    validation_string = json.loads(validation["Payload"].read().decode("utf-8"))["body"]
+    analysis_found_count = json.loads(validation_string)["paging"]["totalItems"]
+
+    if len(analysis_id_list) != analysis_found_count:
+        analysis_confirmed = []
+        analysis_found = {}
+        json_output = json.loads(validation_string)
+
+        for detection in json_output["detections"]:
+            analysis_found[detection["id"]] = detection
+        # Build a list of analysis we've confirmed are in Panther
+        for analysis in analysis_id_list:
+            if analysis not in analysis_found:
+                logging.info("%s was not found, skipping...", analysis)
+            else:
+                analysis_confirmed.append(analysis)
+        return analysis_confirmed
```

Comment:
I think you could potentially get rid of this list and just return the keys from the dict 
```suggestion
        return analysis_found.keys()
```

Reply:
Yup, good idea. Apparently it has to be `list[analysis_found.keys()` because the `.keys()` function returns `dict_keys` object. TIL